buildSrc/src/main/groovy/org/elasticsearch/gradle/test/VagrantFixture.groovy

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.gradle.test
+
+import org.elasticsearch.gradle.vagrant.VagrantCommandTask
+import org.gradle.api.Task
+
+/**
+ * A fixture for integration tests which runs in a virtual machine launched by Vagrant.
+ */
+class VagrantFixture extends VagrantCommandTask implements Fixture {
+
+    private Task stopTask = null
+
+    public VagrantFixture() {
+        project.afterEvaluate {
+            // force the initialization of the stop task at the end
```

Comment:
You'll need to be careful here. afterEvaluate is very tricky for this exact reason.  Calling afterEvaluate inside an afterEvaluate is just wrong.  In this case, I would create the task outside of afterEvalute. You can also override setBoxName and setEnvironmentVars, so that those are copied to the stop task whenever they are changed in the dsl. With setting `args`, I would use a closure to delay evaluation. So instead of `this.boxName`, use `"${-> this.boxName}"`.  I think separately we could also cleanup VagrantCommandTask so that it has a `command` in addition the `boxName`, and then any args are just appended, instead of having to redefine the args for every subclass and needing to use boxName.